src/clj/game/cards/hardware.clj

```diff
@@ -234,6 +234,52 @@
                                 card nil)
                               (rez state :corp eid target))))}]})
 
+(defcard "BMI Buffer"
+  (let [grip-program-trash?
+        (fn [card]
+          (and (runner? card)
+               (program? card)
+               (in-discard? card)
+               (= (first (:previous-zone card)) :hand)))
+        triggered-ability
+        {:async true
+         :effect (req (doseq [c (filter grip-program-trash? (mapv #(find-latest state (:card %)) targets))]
```

Comment:
Doing a `find-latest` here because the card may have moved since the trash event (for example Buffer Drive) - in the case that it's not gone to heap then it shouldn't host on BMI Buffer.